client/api/personalization/use-user-preferences.ts

```diff
@@ -14,54 +14,65 @@ type UserPreferencesResponse = {
 
 const GET_USER_PREFERENCES_QUERY = 'GET_USER_PREFERENCES_QUERY';
 
-function getUserPreferences(fpData: GetResult): Promise<UserPreferencesResponse> {
-  return apiRequest('/api/personalization/get-user-preferences', fpData);
+function getUserPreferencesFromServer(fpData: GetResult, abortSignal: AbortSignal): Promise<UserPreferencesResponse> {
+  return apiRequest('/api/personalization/get-user-preferences', fpData, {}, abortSignal);
 }
 
-function updateUserPreferences(fpData: GetResult, preferences: UserPreferences) {
+function updateUserPreferencesOnServer(fpData: GetResult, preferences: UserPreferences) {
   return apiRequest('/api/personalization/update-user-preferences', fpData, preferences);
 }
 
 export function useUserPreferences() {
   // Get visitorId
   const { data: fingerprintResult } = useVisitorData();
 
+  // An abort controller to cancel the query if the user changes dark mode while the query is still loading
+  const abortController = useMemo(() => new AbortController(), []);
```

Comment:
Shouldn't the abort controller be created on a per-request basis rather than once?

Reply:
That sounds right, but I am not sure it makes a difference in our case. I guess if you have 1 AbortController and pass its signal to multiple requests, you cancel all of them at once by calling `abort()` right? We only have one request we call only once. 

If I create it inside `useQuery`, how would I access it inside `useMutation`? 
<img width="1112" alt="image" src="https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/assets/11249993/08963657-ac1d-4137-ab83-e8e78cac957a">

Refs like this?
```
  // Query database for user preferences for this visitorId
  const { data: preferencesResponse } = useQuery(
    GET_USER_PREFERENCES_QUERY,
    () => {
      const abortController = new AbortController();
      abortControllerRef.current = abortController;
      return getUserPreferencesFromServer(fingerprintResult, abortController.signal);
    },
```
Also, what if you cancel before the query is created/triggered? `abortControllerRef.currrent` is `undefined`, you get an error and the request is not canceled.

Reply:
I see that my suggestion makes it more complicated, so let's just keep it like it is 😁

Reply:
Good thinking anyway, thanks!